code/script.js

```diff
@@ -229,97 +237,171 @@ const setSecret = () => {
+
     } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      No, the person doesn't have ${value} ${category}!`
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+
+  } else if (category === 'eyes') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      Yes, the person has ${value} ${category}!`
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      No, the person doesn't have ${value} ${category}!`
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value) 
     }
-  }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
+  } if (category === 'accessories') {
+    if (keep) {
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      Yes, the person wears ${value}!`
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    
+    } else {
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      No, the person doesn't wear ${value}!`
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }
+  } else if (category === 'other') {
+    if (keep) {
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      Yes, the person is a ${value}!`
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
+    
+    } else {
+      modal.style.display = 'block'
+      modalText.innerHTML = `
+      No, the person is not a ${value}!`
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    }
+  }  
 
-  // Invoke a function to redraw the board with the remaining people.
-}
+  // When the user clicks on close (x), the modal will close
+  span.onclick = function() {
+   modal.style.display = "none";
+  }
+
+  // When the user clicks anywhere outside of the modal, close it
+  window.onclick = function(event) {
+    if (event.target == modal) {
+      modal.style.display = "none";
+    }
+  }
```

Comment:
I saw in a readme that you had problem with the close button at your modal window. My recommendation is to put your modal window in a function so you can call it every  time when you want to show a modal window. Here is my code for modal window function :)
```

const modalWindow = (category, value, keep) => {
      if(category === 'accessories') {
        if(keep) {
          modal.style.display = "block"; // showing modal window to the user
          document.getElementById('modal-text').innerHTML = `Yes, this person wears ${value}` // changing text in the paragraph and showing user
          charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value)) // keeping all characters that has that value
        } else {
          modal.style.display = "block";
          document.getElementById('modal-text').innerHTML = `No, this person doesnt wears ${value}`
          charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value)) // removing all characters that has not the value
        }
      } 
      else if(category === 'other') {
        if(keep) {
          modal.style.display = "block"; // showing modal window to the user
          document.getElementById('modal-text').innerHTML = `Yes, this person does ${value}` // changing text in the paragraph and showing user
          charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
        } else {
          modal.style.display = "block";
          document.getElementById('modal-text').innerHTML = `No, this person doesn't ${value}`
          charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
        }
      }
      else if(category === 'hair' || category === 'eyes') {
        if(keep) {
          modal.style.display = "block"; // showing modal window to the user
          document.getElementById('modal-text').innerHTML = `Yes, this person has ${value} ${category}` // changing text in the paragraph and showing user
          charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
        } else {
          modal.style.display = "block";
          document.getElementById('modal-text').innerHTML = `No, this person hasn't ${value} ${category}`
          charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
        }
      }

  // "close" (x) button that user can use to close the modal window
  span.addEventListener('click', () => {
    modal.style.display = "none";
  })

  // When the user clicks anywhere outside of the modal, close it
  window.onclick = (event) => {
    if (event.target == modal) {
      modal.style.display = "none";
    }
  }

}
```